pkg/build/strategies/sti/sti.go

```diff
@@ -286,7 +286,7 @@ func (b *STI) Exists(config *api.Config) bool {
 
 	// can only do incremental build if runtime image exists, so always pull image
 	previousImageExists, _ := b.docker.IsImageInLocalRegistry(config.Tag)
-	if config.ForcePull {
+	if !previousImageExists || config.ForcePull {
```

Comment:
Actually existing previous image might be older than you'd wanted to be. Let me give you an example from OS world:
I'm doing a build (each time incremental) the first lands on node-1, 2nd on node-2, now if the build 3rd will land on node-1 it'll use artifacts from build-1 which is not the latest (build-2 is). That's why it was done so to always pull that image and not to look at `previousImageExists` flag, at least that's how I remembered it when doing it so.

Or we just don't care about that detail because in the worst case we'll re-download the artifacts that changed between build-1 and build-2?